magic_the_gathering/game_state.py

```diff
@@ -68,18 +76,23 @@ def set_decks(self, decks: List[List[Card]]):
         self.decks = decks
 
     def cast_card(self, player_index, hand_card_index):
+        # TODO: Should we implement a play_card method? Is there a difference between casting and playing a card?
```

Comment:
There is.
- we don't cast lands, we play them
- a permanent being reanimated (so going from the graveyard to the battlefield) isn't cast and that may affect its effect or the effect of other cards, like reanimating this guy https://scryfall.com/card/2x2/1/emrakul-the-aeons-torn
- it is possible to put a permanent from one's hand directly onto the battlefield without casting it https://scryfall.com/card/a25/168/elvish-piper